pages/querying/differences-in-cypher-implementations.mdx

```diff
@@ -0,0 +1,307 @@
+```cypher
+MATCH (:Person {name:'Tom Hanks'})-[r:!ACTED_IN]->(m:Movie)
+Return type(r) AS type, m.title AS movies
+```
+In Memgraph, such a construct is not supported, but there is still a workaround:
+```cypher
+MATCH (p:Person {name:'Tom Hanks'})-[r]->(m:Movie)
+WHERE type(r) != "ACTED_IN"
+RETURN type(r) AS type, m.title AS movies;
+```
+
+### Search for patterns of a fixed length
+
+In Neo4j, to search for patterns of a fixed length, you would use the following construct:
+```cypher
+MATCH (tom:Person {name:'Tom Hanks'})--{2}(colleagues:Person)
+RETURN DISTINCT colleagues.name AS name, colleagues.born AS bornIn
+ORDER BY bornIn
+LIMIT 5
+```
+Memgraph does not support such a construct, but since it has built-in traversals, you can achieve the same with the [depth-first search](/advanced-algorithms/built-in-graph-algorithms#depth-first-search) (DFS) algorithm:
+```cypher
+MATCH (tom:Person {name:'Tom Hanks'})-[*2]-(colleagues:Person)
+RETURN DISTINCT colleagues.name AS name, colleagues.born AS bornIn
+ORDER BY bornIn
+LIMIT 5
+```
+
+Similarly, to match a graph for patterns of a variable length, you would run the following query in Neo4j:
+```cypher
+MATCH (p:Person {name:'Tom Hanks'})--{1,4}(colleagues:Person)
+RETURN DISTINCT colleagues.name AS name, colleagues.born AS bornIn
+ORDER BY bornIn, name
+LIMIT 5
+```
+In Memgraph, again use DFS:
+```cypher
+MATCH (p:Person {name:'Tom Hanks'})-[*1..4]-(colleagues:Person)
+RETURN DISTINCT colleagues.name AS name, colleagues.born AS bornIn
+ORDER BY bornIn, name
+LIMIT 5
+```
+
+### Unsupported constructs
+
+{<h4>CALL subqueries in transactions</h4>}
+Such a query is not supported in Memgraph, but you can use the [periodic](/advanced-algorithms/available-algorithms/periodic) module to execute a query periodically in batches.
+
+{<h4>EXISTS subqueries</h4>}
+Such clause is not supported in Memgraph, but you can use `exists()` [pattern function](/querying/functions#pattern-functions) with the `WHERE` clause to [filter with pattern expressions](/querying/clauses/where#17-filter-with-pattern-expressions). 
+
+The following constructs are not yet supported in Memgraph:
+- **`EXISTS` subquery with `WHERE` clause** -> Track progress on [GitHub](https://github.com/memgraph/memgraph/issues/1483) and add a comment if you require such a feature.
+- **Nesting `EXISTS` subqueries** -> Track progress on [GitHub](https://github.com/memgraph/memgraph/issues/905) and add a comment if you require such a feature.
+- **`EXISTS` subquery outside of a `WHERE` clause** -> Track progress on [GitHub](https://github.com/memgraph/memgraph/issues/1308) and add a comment if you require such a feature.
+
+For all other unsupported constructs that you require, please open an issue on our [GitHub repository](https://github.com/memgraph/memgraph/issues). Learn more in our Community Call on [Subqueries & Patterns in Filtering Clauses](https://www.youtube.com/watch?v=QYFU5d_xLIs&list=PL7Eotag2rRhZssS4f11PKAHuCykMCljg3).
```

Comment:
Maybe this can be general note at the end also?

Reply:
I agree, I'll add it